File: Matlab/SCRoF_Fevrier/as_Crit.m
function [crit,corr]=as_Crit(p,G,combine,cible)
if iscell(G)
    combine=G{2};
    cible=G{3};
    G=G{1};
end
try
[cr,corr]=asCrit(p,G,combine,cible);
catch
    keyboard
end
% [cr,corr]=asCrit(p,varargin(:));
crit=-cr;

File: R0/asCrit.R
asCrit <- function(p, G, combine, cible = NULL, to.opt = TRUE){
  if(is.null(cible)){
    v <- ncol(G)
    cible <- which(!(1:v) %in% combine)
  }
  #p <- mean(p)
  p <- c(p, -1)
  SP <- G[, combine] %*% p
  SP <- SP / c(sqrt(t(SP) %*% SP))
  corr <- t(SP) %*% G
  corr[, setdiff(1:ncol(G), setdiff(cible, combine))] <- 0
  crit <- max(corr^2, na.rm = TRUE)
  if(is.complex(crit)){
    crit <- 99e9 
  }
  if(!to.opt){
    crit <- list(crit = crit, corr = corr)
  }
  crit
}
